packages/common/local-storage/src/forage.ts

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type Signal } from '@preact/signals-core';
+import { type DeepSignal, deepSignal } from 'deepsignal';
+import localforage from 'localforage';
+
+import type { UnsubscribeCallback } from '@dxos/async';
+
+/**
+ * Local storage backed store.
+ * https://developer.mozilla.org/en-US/docs/Web/API/Window/await localforage
+ * DevTools > Application > Local Storage
+ */
+export class LocalForageStore<T extends object> {
+  private async _get(key: string) {
+    const value = await localforage.getItem<any>(key);
+    return value === null ? undefined : value;
+  }
+
+  private async _set(key: string, value: any) {
+    if (value === undefined) {
+      await localforage.removeItem(key);
+    } else {
+      await localforage.setItem(key, value);
+    }
+  }
+
+  private readonly _subscriptions = new Map<string, UnsubscribeCallback>();
+
+  public readonly values: DeepSignal<T>;
+
+  constructor(
+    private readonly _prefix: string,
+    defaults?: T,
+  ) {
+    this.values = deepSignal<T>(defaults ?? ({} as T));
+  }
+
+  // TODO(burdon): Reset method (keep track of binders).
+
+  /**
+   * Binds signal property to local storage key.
+   */
+  async prop<T>(prop: Signal<T | undefined>, lkey: string) {
+    const key = this._prefix + '/' + lkey;
+    if (this._subscriptions.has(key)) {
+      return this;
+    }
+
+    const current = await this._get(key);
```

Comment:
ah I see how the get being async might not be ideal, I was thinking about the set.

This is probably fine for client since all of it's settings require reloads, I'd just leave it there. I'd also mark this class as deprecated with notes about this so it doesn't get used for anything else.

Reply:
I handled it as a special case to not use async API everywhere